backend/src/models/streakCalculation.ts

```diff
@@ -0,0 +1,22 @@
+import { User } from "./user.js";
+
+export async function calculateStreak(userID : string) {
+  const user = await User.findById(userID);
+
+  const lastAnsweredTimestamp = user!.lastAnsweredTimestamp;
+  if (!lastAnsweredTimestamp) return 0;
+
+  // Get today's date
+  const today = new Date();
+  today.setHours(0, 0, 0, 0);
+
+  // Get the last login date
+  const lastLoginDate = new Date(lastAnsweredTimestamp);
+  lastLoginDate.setHours(0, 0, 0, 0);
+
+  // Calculate the difference in days between today and the last login date
+  const timeDifference = today.getTime() - lastLoginDate.getTime();
+  const daysDifference = Math.floor(timeDifference / (1000 * 60 * 60 * 24));
+
+  return daysDifference;
```

Comment:
Would this be the days since their last question answered? We might need to update whenever they answer a question and see if their last question answered was yesterday/today, otherwise reset their streak.

Reply:
I think I fixed it in the latest push